Source/WebCore/dom/Document.cpp

```diff
@@ -10723,19 +10725,51 @@ void Document::flushDeferredRenderingIsSuppressedForViewTransitionChanges()
     }
 }
 
-RefPtr<ViewTransition> Document::startViewTransition(RefPtr<ViewTransitionUpdateCallback>&& updateCallback)
+ExceptionOr<RefPtr<ViewTransition>> Document::startViewTransition(StartViewTransitionCallbackOptions&& callbackOptions)
 {
     if (!globalObject())
         return nullptr;
 
-    Ref viewTransition = ViewTransition::create(*this, WTFMove(updateCallback));
+    if (callbackOptions && std::holds_alternative<StartViewTransitionOptions>(*callbackOptions) && !settings().viewTransitionsTypeEnabled())
+        return Exception { ExceptionCode::TypeError };
```

Comment:
I think it's probably OK to remove this and let people do `{ update: () => {} }` even if the types preference is disabled. The types stuff is properly gated in the IDL.